src/choreo/vote/fd_vote.c

```diff
@@ -1,34 +1,50 @@
 #include "fd_vote.h"
+#include <string.h>
 
 #pragma GCC diagnostic ignored "-Wformat"
 #pragma GCC diagnostic ignored "-Wformat-extra-args"
 
 ulong
 fd_vote_txn_generate(fd_compact_vote_state_update_t* compact_vote_update,
-                     fd_pubkey_t* validator_pubkey,
-                     fd_pubkey_t* vote_acct_pubkey,
-                     uchar* validator_privkey,
-                     uchar* vote_acct_privkey,
+                     fd_pubkey_t* node_pubkey,
+                     fd_pubkey_t* authorized_voter_pubkey,
+                     fd_pubkey_t* vote_acct_addr,
+                     uchar* node_privkey,
+                     uchar* authorized_voter_privkey,
                      uchar* recent_blockhash,
```

Comment:
i think we can abstract the API a bit here. let's make a "voter" type that contains the necessary config -- so it has a `node_pubkey`, etc. so the caller doesnt need to manage passing this all in every time

since the voter is only created once on startup

so you can rename `fd_vote` to `fd_voter`, corresponding files etc.

Reply:
Cool, I have created the `fd_voter_t` type and simplified the `fd_vote_txn_generate` interface. I have renamed it to `fd_voter` instead of `fd_vote`. I have also removed all `memcpy` of private keys -- the argument we now pass to `fd_ed25519_sign` is exactly the pointer returned by `fd_keyload_load`, a pointer in the protected memory region.